twisted/plugins/multichain_crawler_plugin.py

```diff
@@ -0,0 +1,89 @@
+from Tribler.dispersy.dispersy import Dispersy
+from Tribler.dispersy.endpoint import StandaloneEndpoint
+
+if os.environ.get('HOME'):
+    tribler_dir = os.path.join(unicode(os.environ.get('HOME')), u'.multichain')
+else:
+    tribler_dir = u'.multichain'
+
+
+class Options(usage.Options):
+    optParameters = [
+        ["statedir", "s", tribler_dir, "Use an alternate statedir"                                    , unicode],
+        ["ip"      , "i", "0.0.0.0" ,  "Dispersy uses this ip"                                        , str],
+        ["port"    , "p", 6421      ,  "Dispersy uses this UDP port"                                  , int],
+    ]
+
+
+class MultichainCrawlerServiceMaker(object):
+    implements(IServiceMaker, IPlugin)
+    tapname = "multichain_crawler"
+    description = "A MultichainCommunity crawler"
+    options = Options
+
+    def makeService(self, options):
+        tracker_service = MultiService()
+        tracker_service.setName("Multichain Crawler")
+
+        def run():
+            # setup
+            if os.path.exists(os.path.join(options["statedir"], "logger.conf")):
+                logging.config.fileConfig(os.path.join(options["statedir"], "logger.conf"))
+            elif os.path.exists("logger.conf"):
+                logging.config.fileConfig("logger.conf")
+            else:
+                logging.basicConfig(format="%(asctime)-15s [%(levelname)s] %(message)s", level=logging.DEBUG)
+            logger = logging.getLogger(__name__)
+
+            crypto = ECCrypto()
+            dispersy = Dispersy(StandaloneEndpoint(options["port"], options["ip"]),
+                                options["statedir"],
+                                u'dispersy.db',
+                                crypto)
+            if not dispersy.start():
+                raise RuntimeError("Unable to start Dispersy")
+            master_member = dispersy.get_member(
+                public_key='3081a7301006072a8648ce3d020106052b810400270381920004059f45b75d63f865e3c7b350bd3ccdc99dbfbf'
+                           '76fdfb524939f0702233ea9ea5d0536721cf9afbbec5693798e289b964fefc930961dfe1a7f71c445031434aba'
+                           '637bb93b947fb81603f649d4a08e5698e677059b9d3a441986c16f8da94d4aa2afbf10fe056cd65741108fe6a8'
+                           '80606869ca81fdcb2db302ac15905d6e75f96b39ccdaf068bdbbda81a6356f53f7ce4e'.decode("HEX"))
+            my_member = dispersy.get_member(private_key=crypto.key_to_bin(crypto.generate_key(u"curve25519")))
+            MultiChainCommunityCrawler.init_community(dispersy, master_member, my_member)
+
+            self._stopping = False
+
+            def signal_handler(sig, frame):
+                logger.info("Received signal '%s' in %s (shutting down)" % (sig, frame))
+                if not self._stopping:
+                    self._stopping = True
+                    try:
+                        dispersy.stop()
+                    except Exception, e:
```

Comment:
General try except, do you know what exceptions can be raised during the stop? Catching those will be cleaner.